.github/workflows/publish-artifacts.yml

```diff
@@ -0,0 +1,173 @@
+name: Publish artifacts to ORAS
+
+on:
+  push:
+    branches:
+    - main
+
+env:
+  RUST_TOOLCHAIN: 1.80.0
+
+jobs:
+  publish-aa:
+    permissions:
+      contents: read
+      packages: write
+      id-token: write
+    strategy:
+      matrix:
+        tee:
+        - none
+        - amd
+        - az-cvm-vtpm
+        - tdx
+        - se
+        - cca
+        include:
+        - tee: none
+          libc: musl
+          arch: x86_64
+        - tee: amd
+          libc: musl
+          arch: x86_64
+        - tee: az-cvm-vtpm
+          libc: gnu
+          arch: x86_64
+        - tee: tdx
+          libc: gnu
+          arch: x86_64
+        - tee: se 
+          libc: gnu
+          arch: s390x
+        - tee: cca
+          libc: musl
+          arch: x86_64
+    runs-on: ${{ matrix.arch == 's390x' && 's390x' || 'ubuntu-22.04' }}
+    env:
+      TEE_PLATFORM: ${{ matrix.tee }}
+      LIBC: ${{ matrix.libc }}
+      REGISTRY: ghcr.io
+      IMAGE_NAME: ${{ github.repository }}
+    steps:
+    - name: Take a pre-action for self-hosted runner
+      if: matrix.arch == 's390x'
+      run: |
+        if [ -f "${HOME}/script/pre_action.sh" ]; then
+          "${HOME}/script/pre_action.sh" cc-guest-components
+        fi
```

Comment:
@BbolroC can confirm, but I think we use the `ACTIONS_RUNNER_HOOK_JOB_STARTED` and `ACTIONS_RUNNER_HOOK_JOB_COMPLETED` envs on the s390x runner now rather than needing to call the pre/post install action scripts directly.

Reply:
so this step can be removed?

Reply:
Yeah, that's my understanding

Reply:
FYI: I've found https://github.com/confidential-containers/trustee/pull/453 which shows this is implemented in the coco org.